### PR TITLE
feat(e2e): add e2e tests for the slashing module

### LINKED ISSUE #1679: Add Slashing module E2E tests
____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned
- [ ] Is a spike necessary to map out how the issue should be approached?

### PR DESCRIPTION
**Pull request by Pantani**
close #1679

## Description

Add tests for the slashing module.

### Test workflow

- Create a jailed keys and the validator into the genesis JSON;
- Iterate over all validators and find the jailed one;
- Execute the Unjail gaiad command for the jailed validator;
- Query the validator and check if is already unjailed;

### CODE REVIEW

File: tests/e2e/query.go
Comment (glnro) [inline tests/e2e/query.go:18]: This is a duplicate of the line above

File: tests/e2e/query.go
Comment (glnro) [inline tests/e2e/query.go]: Up to you but perhaps we could start to use [this](https://github.com/cosmos/gaia/blob/main/tests/e2e/http_util.go#L9) more in our REST queries

**Comment by mmulji-ic** [comment]
Still an issue with the e2e test:
Test:       	TestIntegrationTestSuite/TestSendTokensFromNewGovAccount/vote_gov_proposal#01
panic: invalid coin denoms: , uatom

File: tests/e2e/e2e_setup_test.go
Comment (okwme) [inline tests/e2e/e2e_setup_test.go]:
```suggestion
	jailedValAcc, err := jailedValKey.GetAddress()
```

since there's jailed and non-jailed vals getting added in the same function would be good to keep a var name that distinguishes them as such. I was confused for a moment below adding the delegations to the genState whether this was actually the jailed validator or not.

File: tests/e2e/e2e_setup_test.go
Comment (okwme) [inline tests/e2e/e2e_setup_test.go]:
```suggestion
	jailedValAddr := sdk.ValAddress(jailedValAcc)
```

File: tests/e2e/e2e_setup_test.go
Comment (okwme) [inline tests/e2e/e2e_setup_test.go]:
```suggestion
		jailedValAddr,
```

File: tests/e2e/e2e_setup_test.go
Comment (okwme) [inline tests/e2e/e2e_setup_test.go]:
```suggestion
		DelegatorAddress: jailedValAcc.String(),
```

File: tests/e2e/e2e_setup_test.go
Comment (okwme) [inline tests/e2e/e2e_setup_test.go]:
```suggestion
		ValidatorAddress: jailedValAddr.String(),
```

File: tests/e2e/e2e_setup_test.go
Comment (okwme) [inline tests/e2e/e2e_setup_test.go]:
```suggestion
	baseJailedAccount := authtypes.NewBaseAccount(jailedValAcc, pubKey, 0, 0)
```

File: tests/e2e/e2e_setup_test.go
Comment (okwme) [inline tests/e2e/e2e_setup_test.go]:
```suggestion
		Address: jailedValAcc.String(),
```

**Review by okwme — APPROVED** [review]
Left a group of optional suggestions about naming clarity.
It's a suggestion but not a strong one, if you feel like the current naming has justification feel free to merge without changing. Otherwise feel free to merge after changing without waiting for another review.

Nice job on the small code cleanliness changes along the way, really appreciate leaving things nicer than you found them : )

**Pull request merged**
